src/Parser/Shaman/Elemental/Modules/Items/SmolderingHeart.js

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import ITEMS from 'common/ITEMS';
+import SPELLS from 'common/SPELLS';
+import ItemIcon from 'common/ItemIcon';
+import ItemLink from 'common/ItemLink';
+import { formatPercentage } from 'common/format';
+
+class SmolderingHeart extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+  };
+
+  SmolderingProcs = 0;
+  SmolderingDuration = 0;
+  MaelstromSpent = 0;
+  AscendanceCastEventIds = new Set();
+
+  smoldering_uptime() {
+    return this.SmolderingDuration / this.owner.fightDuration;
+  }
+
+  expected_smoldering_uptime() {
+    // Smoldering Heart (Elemental) gives a 0.10% (0.001 decimal) chance per point of Maelstrom spent to grant Ascendance for 10 seconds (10000ms).
+    const EXPECTED_PROC_COUNT = this.MaelstromSpent * 0.001;
+    const EXPECTED_UPTIME = (EXPECTED_PROC_COUNT * 10000) / this.owner.fightDuration;
+    return EXPECTED_UPTIME;
```

Comment:
Would be nice if you extracted the magic numbers outside the class as constants.
```jsx
const SMOLDERING_HEART_CHANCE = 0.001;
const SMOLDERING_HEART_ASCENDANCE_DURATION = 10000;
```
Also the variables here should be lowercase - `expectedProcCount`, `expectedUptime`.